go-controller/pkg/util/util.go

```diff
@@ -55,6 +54,23 @@ func StringArg(context *cli.Context, name string) (string, error) {
 	return val, nil
 }
 
+// GetIPNetFullMask returns an IPNet object for IPV4 or IPV6 address with a full subnet mask
+func GetIPNetFullMask(ipStr string) (*net.IPNet, error) {
+	ip := net.ParseIP(ipStr)
+	if ip == nil {
+		return nil, fmt.Errorf("failed to parse IP %q", ipStr)
+	}
+	mask := GetIPFullMask(ip)
+	// mask will be replaced
+	ip, ipNet, err := net.ParseCIDR(fmt.Sprintf("%s/32", ipStr))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse IP %q: %v", ipStr, err)
+	}
```

Comment:
why do you parse here? you can just build the IPNet directly

Reply:
True - fixing.